src/react-components/room/VoiceButtonContainer.js

```diff
@@ -1,15 +1,29 @@
 import React, { useEffect, useRef } from "react";
 import PropTypes from "prop-types";
-import { ReactComponent as MicrophoneIcon } from "../icons/Microphone.svg";
-import { ReactComponent as MicrophoneMutedIcon } from "../icons/MicrophoneMuted.svg";
-import { ToolbarButton } from "../input/ToolbarButton";
 import { useMicrophone } from "./useMicrophone";
-import { FormattedMessage } from "react-intl";
+import { AudioToolbarPopoverButton } from "./AudioToolbarPopover";
+import { useSpeakers } from "./useSpeakers";
+import { useSound } from "./useSound";
+import { SOUND_SPEAKER_TONE } from "../../systems/sound-effects-system";
 
 export function VoiceButtonContainer({ scene }) {
   const buttonRef = useRef();
 
-  const { isMicMuted, micVolume, toggleMute, isMicEnabled } = useMicrophone(scene);
+  const {
+    micVolume,
```

Comment:
Oof, I didn't catch this when reviewing the PR when `useMicrophone` was added but I bet this is causing this component (and all its children) to constantly re-render. Also since this hook is here we have the interval always running, not just when the popover is open. I bet that's non trivial perf wise. 

Simple first step fix would be to split out `useVolumeMeter` and then only call it in the most deeply nested place its being used. That would at least only have the react renders going when actually needed (entry flow and when audio popup is going). That might be enough.

Second stage optimization would be to have it mutate a ref directly rather than going through react state/render. Something like:
```js
useVolumeMeter(function(volume) {
  ref.current.style.clipPath = ...
})
```

Only noticed this now because I have been thinking about react-three-fiber and having that work well is all about not doing any of the gameloop stuff through actual react state.

Reply:
Yep, probably this would never be an issue in this specific case as we are constantly updating but it's good to know and get used to it. I've updated to use a callback ref.